src/gto-cas-login.php

```diff
@@ -1,509 +1,539 @@
+    /**
+     * Prevent cloning of the object
+     */
+    private function __clone() {
     }
 
-    const TEXT_DOMAIN = 'gtocas';
-
-    const VERSION = '0.1';
-
-    class CASLogin {
-        /**
-         * Singleton instance
-         * @var CASLogin
-         */
-        private static $instance;
-
-        /**
-         * Returns the CASLogin singleton
-         * @return CASLogin
-         */
-        public static function singleton() {
-            if ( ! isset( self::$instance ) ) {
-                $class          = __CLASS__;
-                self::$instance = new $class();
-            }
-            return self::$instance;
-        }
+    /**
+     * CAS client mode
+     * @var string
+     */
+    const CLIENT = 'client';
 
-        /**
-         * Prevent cloning of the object
-         */
-        private function __clone() {
-        }
+    /**
+     * CAS proxy mode
+     * @var string
+     */
+    const PROXY = 'proxy';
 
-        /**
-         * CAS client mode
-         * @var string
-         */
-        const CLIENT = 'client';
+    const USER_META_GUID = 'guid';
 
-        const USER_META_GUID = 'guid';
+    const CAS_ATTRIBUTE_GUID = 'ssoGuid';
+    const CAS_ATTRIBUTE_FIRST_NAME = 'firstName';
+    const CAS_ATTRIBUTE_LAST_NAME = 'lastName';
 
-        const CAS_ATTRIBUTE_GUID       = 'ssoGuid';
-        const CAS_ATTRIBUTE_FIRST_NAME = 'firstName';
-        const CAS_ATTRIBUTE_LAST_NAME  = 'lastName';
+    public $base_uri;
 
-        public $base_uri;
+    /**
+     *
+     * @var \CAS_Client
+     */
+    private $_cas_client;
 
-        /**
-         *
-         * @var \CAS_Client
-         */
-        private $_cas_client;
+    public $options = array(
+      'hostname' => 'thekey.me',
+      'port'     => 443,
+      'uri'      => 'cas',
+    );
 
-        public $options = array(
-            'hostname' => 'thekey.me',
-            'port'     => 443,
-            'uri'      => 'cas',
-        );
+    protected function __construct() {
+      $this->base_uri = plugin_dir_url( __FILE__ );
 
-        protected function __construct() {
-            $this->base_uri = plugin_dir_url( __FILE__ );
+      $this->register_actions_filters();
+    }
 
-            $this->register_actions_filters();
-        }
+    private function register_actions_filters() {
+      add_action( 'muplugins_loaded', array( &$this, 'initialize_phpcas' ), 5, 0 );
+      add_action( 'activate_header', array( &$this, 'user_activate' ), 10, 0 );
+
+      //Remove the Add New submenu from the Users menu
+      add_action( 'admin_menu', array( &$this, 'remove_add_user_submenu' ), 15 );
+      add_filter( 'current_screen', array( &$this, 'redirect_on_add_user_screen' ), 10, 1 );
+      add_action( 'admin_menu', function () {
+        add_users_page( __( 'Add Users', TEXT_DOMAIN ), __( 'Add Users', TEXT_DOMAIN ), 'manage_options', 'add-user', array(
+          &$this,
+          'add_user_page'
+        ) );
```

Comment:
guessing you auto-formatted the file, do you want to keep the `array( &$this, 'add_user_page' )` callback multiline here, or collapse it back down to a single line

Reply:
I'll reformat it slightly.